### ISSUE TITLE
Grafana package can no longer be installer from apt repo https://apt.grafana.com/

### ISSUE THREAD

**Issue by VGerris** [grafana/grafana-ansible-collection#182]
Since today, using the role fails with : 

```bash
TASK [grafana.grafana.grafana : Install Grafana] ****************************************************************************************************************************
fatal: [icinga2test.binero.net]: FAILED! => {"changed": false, "msg": "No package matching 'grafana' is available"}
```
If I look in the packages available, I see grafana-nightly only.
The content of my apt file is :

```bash
root@icinga-server:~# cat /etc/apt/sources.list.d/apt_grafana_com.list 
deb [arch=amd64 signed-by=/usr/share/keyrings/grafana.key] https://apt.grafana.com/ stable main
```
Did something change in the repo ?
This causes my playbook that worked last week to fail.

The install tasks has : **grafana_package** as a variable but it does not work to use that in the playbook.

So the apt is matching the current documentation.

As a workaround I added the following in an Ansible task:

```yaml
    - name: Install grafana via deb because apt repo misses package
      ansible.builtin.apt:
        deb: https://dl.grafana.com/oss/release/grafana_10.4.1_amd64.deb
```
Taken from : https://grafana.com/grafana/download?edition=oss .
This works as a workaround because the package does not need re-installation.

I hope this can be corrected so the playbook works again.
It may be a good idea to implement the **grafana_package** variable if more than 1 package can be installed, so it can be chosen by parameter.

**Comment by VGerris** [comment]
Thank you for the quick response. The work around only works if at the time of restart the config allows grafana to start, since the playbook checks it.

**Comment by gardar** [comment]
The issue is with the repository, not the ansible role, there are other users reporting the same issue today on the grafana forums.

https://community.grafana.com/t/grafana-package-cannot-be-found-apt/119021/4
https://community.grafana.com/t/unable-to-install-grafana-from-apt-repository-on-debian-bookworm/119040
And there seems to have been a similar issue with the repo last week: https://status.grafana.com/incidents/xblqfhgkz0sg

@ishanjainn perhaps you can bring this to the attention of your colleagues at grafana?

**Comment by VGerris** [comment]
One can argue if a package not being available is part of the playbook, it does break the playbook. Could be enough to track it. Anyway, the issue seems resolved at the moment, so I will close it. Thank you!

**Comment by gardar** [comment]
> One can argue if a package not being available is part of the playbook, it does break the playbook. Could be enough to track it.

Well the same argument could apply to just about any ansible role, but it's highly unconventional to have repos that suddenly have missing packages so I don't feel like we should put in additional code in the role for a scenario such as this.

**Comment by VGerris** [comment]
I agree, I did that in my own playbook only that uses the role. I added a
script that fetches latest version and downloads the deb if not in repo.
Had to change a lot to make that work, but kind of useful to have now :).
Not expecting this to happen again actually and it's also solvable by
downloading manually so agree with no mods needed. Although it could be
nice to have that var I mentioned to install nightly for example? I'm happy
it's solved, thank you!
